src/classes/CRLP_Batch_Base.cls

```diff
@@ -335,224 +350,348 @@ public abstract class CRLP_Batch_Base {
+         * start query for Non Skew jobs.
+         * @return The query string.
+         */
+        String buildQueryForNonSkewJob(){
+            // Build a query string of the target Summary object.
+            String query = CRLP_Query_SEL.buildObjectQueryForRollup(base.summaryObjectType);
+
+            // Fields in the fieldsToCheckForNonZero[] list must always be in the Contact query
+            if (fieldsToCheckForNonZero != null) {
+                for (String f : fieldsToCheckForNonZero) {
+                    if (!query.containsIgnoreCase(f)) {
+                        query = query.split(' FROM ')[0] + ', ' + f + ' FROM ' + query.split(' FROM ')[1];
+                    }
+                }
             }
 
-            when ContactSoftCredit {
-                whereClause = 'Id IN (SELECT ContactId FROM OpportunityContactRole)';
+            List<String> whereClauses = new List<String>();
+
+            // Add additional query filters as needed
+            if (base.recordIds != null && base.recordIds.size() > 0) {
+                whereClauses.add('Id IN :recordIds');
+            } else {
+                // Exclude Accounts or Contacts that will be rolled up in the SkewMode batch job
+                String skewClause = CRLP_RollupBatch_SVC.getSkewWhereClause(base.jobType, CRLP_RollupProcessingOptions.BatchJobMode.NonSkewMode);
+                if (!String.isEmpty(skewClause)) {
+                    whereClauses.add(skewClause);
+                }
+
+                if (lastIdProcessed != null) {
+                    whereClauses.add('Id > \'' + lastIdProcessed + '\'');
+                }
+
+                String innerJoinWhereClause = getMainQueryInnerJoinFilter();
+                if (innerJoinWhereClause != null) {
+                    whereClauses.add(innerJoinWhereClause);
+                }
             }
 
-            when AccountContactSoftCredit, AccountSoftCredit {
-                // It's not possible to do an inner query on this due to the following Salesforce error:
-                // "The inner select field 'Contact.AccountId' cannot have more than one level of relationships"
+            if (!whereClauses.isEmpty()) {
+                query += ' WHERE ' + String.join(whereClauses, ' AND ');
             }
 
-            when RecurringDonations {
-                whereClause = 'Id IN (SELECT npe03__Recurring_Donation__c FROM Opportunity)';
+            // If there are no active rollups for the object type, then force the query to return zero records.
+            if (base.hasActiveRollupsForThisJob() == false) {
+                // This is needed because can't return null as a valid iterator.
+                query += ' LIMIT 0';
+            } else if (base.recordIds == null) {
+                if (chunkSize != null) {
+                    // In an LDV Org, there may be too many records to query in the main query. Core support has recommended
```

Comment:
Super minor comment, but can we keep these comments on one line?

Reply:
I'd say this is safe to remove now anyway.  I removed it but Mike let me know if we should keep it.

Reply:
I do like the comments just to explain the "why" behind chunk mode, but it's also not critical to be there. What I might suggest is to put this particular comment (that explains the why) into the apexDoc comment of the `isChunkModeEnabled` method.